### PR TITLE
Replace package object compiletime by top-level definitions; move compiletime.S to compiletime.ops.int.S

### PR DESCRIPTION
**Pull request by smarter**
[test_non_bootstrapped]

### CODE REVIEW

File: library/src-bootstrapped/scala/compiletime/package.scala [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+  error("Compiler bug: `constValue` was not evaluated by the compiler")
+
+/**
+ * Use this type to widen a self-type to a tuple. E.g.
+ * ```
+ * val x: (1, 3) = (1, 3)
+ * val y: Widen[x.type] = x
+ * ```
+ */
+type Widen[Tup <: Tuple] <: Tuple = Tup match {
```
Comment (bishabosha) [inline library/src-bootstrapped/scala/compiletime/package.scala:105]: this is meant to be removed now right?

Comment (smarter) [reply library/src-bootstrapped/scala/compiletime/package.scala:105]: oops, rebase error.

**Review by MaximeKjaer — COMMENTED** [review]
If `S` moves, we need to update the documentation here: https://dotty.epfl.ch/docs/reference/metaprogramming/inline.html#constvalue-constvalueopt-and-the-s-combinator

**Comment by smarter** [comment]
Thanks, I updated the doc (and now I'm also wondering if there's really a usecase for `S` instead of  just using` + 1` ?)

**Comment by MaximeKjaer** [comment]
I don't really know if there is a use case. One thing I think `S` can do which `+` cannot is the following:

```scala
import scala.compiletime.S

type MinusOne[N <: Int & Singleton] = N match {
  case S[n] => n
}

@main def test: Unit = {
  val x: MinusOne[3] = 2
}
```

Though of course `MinusOne` could also be implemented as `N - 1` now, instead of as a match type. I don't really know if there are edge cases where the two might be slightly different.

**Comment by smarter** [comment]
OK, sounds like S is still useful until/unless + becomes more powerful.

**Review by MaximeKjaer — APPROVED** [review]

**Review by bishabosha — APPROVED** [review]
LGTM
